static/app/components/nav/sidebar.tsx

```diff
@@ -89,37 +93,85 @@ const SidebarItemList = styled('ul')`
   }
 `;
 
-function SidebarItem({item}: {item: NavSidebarItem}) {
-  const location = useLocation();
-  const isActive = isNavItemActive(item, location);
-  const isSubmenuActive = isSubmenuItemActive(item, location);
-  const _to = resolveNavItemTo(item);
-  const linkProps = _to ? makeLinkPropsFromTo(_to) : {to: '#'};
+interface SidebarItemProps {
+  item: NavSidebarItem;
+}
+
+function SidebarItem({item}: SidebarItemProps) {
+  const to = resolveNavItemTo(item);
+  const Component = to ? SidebarLink : SidebarOverlay;
 
   const FeatureGuard = item.feature ? Feature : Fragment;
   const featureGuardProps: any = item.feature ?? {};
 
   return (
     <FeatureGuard {...featureGuardProps}>
       <SidebarItemWrapper>
-        <SidebarLink
-          {...linkProps}
-          aria-current={isActive ? 'page' : undefined}
-          aria-selected={isActive || isSubmenuActive}
-        >
-          <InteractionStateLayer hasSelectedBackground={isActive || isSubmenuActive} />
+        <Component item={item} key={item.label}>
           {item.icon}
           <span>{item.label}</span>
-        </SidebarLink>
+        </Component>
       </SidebarItemWrapper>
     </FeatureGuard>
   );
 }
 
-const SidebarLink = styled(Link)`
+const StyledLink = styled(Link)`
   position: relative;
 `;
 
+const StyledButton = styled('button')`
+  border: none;
+  position: relative;
+
+  ${linkStyles}
+`;
+
+function SidebarLink({children, item}: PropsWithChildren<SidebarItemProps>) {
+  const location = useLocation();
+  const isActive = isNavItemActive(item, location);
+  const isSubmenuActive = isSubmenuItemActive(item, location);
+  const to = resolveNavItemTo(item);
+  const linkProps = makeLinkPropsFromTo(to!);
+
+  return (
+    <StyledLink
+      {...linkProps}
+      className={isActive || isSubmenuActive ? 'active' : undefined}
+      aria-current={isActive ? 'page' : undefined}
+    >
+      <InteractionStateLayer hasSelectedBackground={isActive || isSubmenuActive} />
+      {children}
+    </StyledLink>
+  );
+}
+
+function SidebarOverlay({children, item}: PropsWithChildren<SidebarItemProps>) {
+  const {isOpen, triggerProps, overlayProps} = useOverlay({
+    isDismissable: true,
+    shouldApplyMinWidth: false,
+    shouldCloseOnBlur: true,
+    position: 'right-end',
+    offset: 0,
+  });
+
+  return (
+    <Fragment>
+      <StyledButton {...triggerProps} className={isOpen ? 'active' : undefined}>
+        <InteractionStateLayer hasSelectedBackground={isOpen} />
+        {children}
+      </StyledButton>
+      {isOpen && (
+        <OverlayMenu {...overlayProps}>
+          <Suspense fallback={<LoadingIndicator />}>
+            {item.overlay ? <item.overlay /> : null}
+          </Suspense>
+        </OverlayMenu>
+      )}
+    </Fragment>
```

Comment:
Why can't this whole thing just use our `DropdownMenu` component?

https://github.com/getsentry/sentry/blob/master/static/app/components/dropdownMenu/index.tsx

Reply:
Migrated this in [e69afcc](https://github.com/getsentry/sentry/pull/79720/commits/e69afcc68bdf675e2f1f6044e054ad9573d6cf6b)